src/extension/infer.rs

```diff
@@ -0,0 +1,855 @@
+                                    *m,
+                                    combined_meta,
+                                    solution.clone(),
+                                    existing_solution.clone(),
+                                ));
+                            }
+                        }
+                        None => {
+                            self.solved.insert(combined_meta, solution.clone());
+                        }
+                    }
+                }
+                if self.variables.contains(m) {
+                    self.variables.insert(combined_meta);
+                }
+                self.shunted.insert(*m, combined_meta);
+            }
+            // This doesn't do anything, but if it did it would just be
+            // implementing the same mechanism as provided by `resolve`
+            let mut updates = HashMap::<(Node, Direction), Meta>::new();
+            for (loc, meta) in self.extensions.iter() {
+                // If there's any node in this CC which equals `meta`, they are
+                // all equal to meta
+                if cc.iter().filter(|cc_meta| meta == *cc_meta).count() > 0 {
+                    updates.insert(*loc, combined_meta);
+                }
+            }
+        }
+        Ok((new_metas, merged))
+    }
+
+    /// Inspect the constraints of a given metavariable and try to find a
+    /// solution based on those.
+    /// Returns whether a solution was found
+    fn solve_meta(&mut self, meta: Meta) -> Result<bool, InferExtensionError> {
+        let mut solved = false;
+        for c in self.get_constraints(&meta).unwrap().clone().iter() {
+            match c {
+                // Just register the equality in the EqGraph, we'll process it later
+                Constraint::Equal(other_meta) => {
+                    self.eq_graph.register_eq(meta, *other_meta);
+                }
+                Constraint::Plus(r, other_meta) => {
+                    if let Some(rs) = self.get_solution(other_meta) {
+                        let mut rrs = rs.clone();
+                        rrs.insert(r);
+                        match self.get_solution(&meta) {
+                            // Let's check that this is right?
+                            Some(rs) => {
+                                if rs != &rrs {
+                                    return Err(self.report_mismatch(
+                                        meta,
+                                        *other_meta,
+                                        rs.clone(),
+                                        rrs,
+                                    ));
+                                }
+                            }
+                            None => self.add_solution(meta, rrs),
+                        };
+                        solved = true;
+                    } else {
+                        // TODO: Try and go backwards with a `Minus` constraint
```

Comment:
are we likely to run in to this case often?

Reply:
I don't think it's so uncommon that we can ignore it. If, in the `dangling_src` test, the `src` node was wired into a node that adds extension requirements rather than leaving them be, we'd hit this case and fail to solve its requirements

Reply:
Fixed it